rmf_fleet_adapter/src/lift_supervisor/Node.cpp

```diff
@@ -92,31 +101,27 @@ void Node::_adapter_lift_request_update(LiftRequest::UniquePtr msg)
 }
 
 //==============================================================================
-void Node::_lift_state_update(LiftState::UniquePtr msg)
+void Node::_lift_state_update(LiftState::UniquePtr state)
 {
   auto& lift_request = _active_sessions.insert(
-    std::make_pair(msg->lift_name, nullptr)).first->second;
+    std::make_pair(state->lift_name, nullptr)).first->second;
 
   if (lift_request)
   {
-    if ((lift_request->destination_floor != msg->current_floor) ||
-      (lift_request->door_state != msg->door_state))
-      lift_request->request_time = this->now();
-    _lift_request_pub->publish(*lift_request);
-    RCLCPP_INFO(
-      this->get_logger(),
-      "[%s] Published lift request to [%s] from lift supervisor",
-      msg->session_id.c_str(), lift_request->destination_floor.c_str()
-    );
+    if ((lift_request->destination_floor != state->destination_floor) ||
```

Comment:
do we also need to check if the mode in the lift request is the same as in the state?

Reply:
Good point, although I think the mode has a risk of being an unreliable indicator across different lifts. We anticipate that some lifts might not have different discrete modes, so their lift adapters might not broadcast the mode in a way that's consistent with our expectations.

Instead [70eddc8](https://github.com/open-rmf/rmf_ros2/pull/418/commits/70eddc8903a560b6cde2c86d358ad2a768c18557) checks to make sure that the `session_id` matches between that state and the current request. We do expect the lift adapters to broadcast the correct session ID once a session has been locked in for a request.